src/Elastic.Apm/Report/Serialization/MetricSetConverter.cs

```diff
@@ -13,25 +16,86 @@ internal class MetricSetConverter : JsonConverter<MetricSet>
 	{
 		public override void WriteJson(JsonWriter writer, MetricSet value, JsonSerializer serializer)
 		{
-			var metrics = new JObject();
-			var samples = new JObject();
+			writer.WriteStartObject();
+			writer.WritePropertyName("samples");
+			writer.WriteStartObject();
 
+			var addedKeys = new HashSet<string>();
 			foreach (var item in value.Samples)
 			{
-				var valueObj = new JObject { { "value", item.KeyValue.Value } };
-
-				if (!samples.ContainsKey(item.KeyValue.Key))
-					samples.Add(item.KeyValue.Key, valueObj);
+				if (addedKeys.Add(item.KeyValue.Key))
+				{
+					writer.WritePropertyName(item.KeyValue.Key
+						.Replace('*', '_')
+						.Replace('"', '_'));
```

Comment:
This should be sanitised according to the spec: https://github.com/elastic/apm-server/blob/7cf56b584c9ff94153018afd1758f27737628873/docs/spec/metricsets/metricset.json#L11-L22